File: backend/app/models/character.rb
class Character < ApplicationRecord
  default_scope { order(name: :asc) }

  def self.ransackable_attributes(_auth_object = nil)
    %w[alias_names animagus blood_status boggart born died eye_color family_members gender hair_color height house jobs
       marital_status name nationality patronus romances skin_color species titles wands weight ]
  end
end

File: frontend/src/types/Character.ts
export default interface Character {
  id: string;
  attributes: {
    slug: string;
    alias_names: string[];
    animagus: string | null;
    blood_status: string | null;
    boggart: string | null;
    born: string | null;
    died: string | null;
    eye_color: string | null;
    family_members: string[];
    gender: string | null;
    hair_color: string | null;
    height: string | null;
    house: string | null;
    image: string | null;
    jobs: string[];
    marital_status: string | null;
    name: string;
    nationality: string | null;
    patronus: string | null;
    romances: string[];
    skin_color: string | null;
    species: string | null;
    titles: string[];
    wands: string[];
    weight: string | null;
    wiki: string;
  };
}
